docs/src/userguide/language_basics.rst

```diff
@@ -88,7 +88,8 @@ define global C variables.
 
             def func():
                 cdef int i, j, k
-                cdef float f, g[42], *h
+                cdef float f, *h
+                cdef float[42] g
```

Comment:
There is still a warning:

```
warning: test.pyx:3:18: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
```

```suggestion
                cdef float f
                cdef float[42] g
                cdef float *h
                # cdef float f, g[42], *h  # mix of pointers, arrays and values in a single line is deprecated
```

In addition, it would be better to explicitly mention the (soft-deprecated) C style declaration and Java style declaration in the documentaion.